src/storage/disk_manager.c

```diff
@@ -4680,19 +4679,21 @@ disk_unreserve_ordered_sectors_without_csect (THREAD_ENTRY * thread_p, DB_VOLPUR
   context.purpose = purpose;
 
   /* note: vsids are ordered */
-  for (start_index = 0; start_index < nsects; start_index = end_index)
+  anchor = 0;
+  for (index = 0; index < nsects - 1; index++)
     {
-      assert (volid < vsids[start_index].volid);
-      volid = vsids[start_index].volid;
-      for (end_index = start_index + 1; end_index < nsects && vsids[end_index].volid == volid; end_index++)
+      assert (volid < vsids[index].volid);
+      volid = vsids[index].volid;
+      if (volid != vsids[index + 1].volid || index == nsects - 2)
```

Comment:
if nsects == 3 and every volid is same then we lost 1 sector (last one). And I think you have to check another scenario.